packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart

```diff
@@ -102,18 +117,61 @@ class AndroidWebViewController extends PlatformWebViewController {
         }
       };
     }),
-    onShowFileChooser: withWeakReferenceTo(this,
-        (WeakReference<AndroidWebViewController> weakReference) {
-      return (android_webview.WebView webView,
-          android_webview.FileChooserParams params) async {
-        if (weakReference.target?._onShowFileSelectorCallback != null) {
-          return weakReference.target!._onShowFileSelectorCallback!(
-            FileSelectorParams._fromFileChooserParams(params),
-          );
-        }
-        return <String>[];
-      };
-    }),
+    onShowFileChooser: withWeakReferenceTo(
+      this,
+      (WeakReference<AndroidWebViewController> weakReference) {
+        return (android_webview.WebView webView,
+            android_webview.FileChooserParams params) async {
+          if (weakReference.target?._onShowFileSelectorCallback != null) {
+            return weakReference.target!._onShowFileSelectorCallback!(
+              FileSelectorParams._fromFileChooserParams(params),
+            );
+          }
+          return <String>[];
+        };
+      },
+    ),
+    onPermissionRequest: withWeakReferenceTo(
+      this,
+      (WeakReference<AndroidWebViewController> weakReference) {
+        return (_, android_webview.PermissionRequest request) async {
+          final void Function(PlatformWebViewPermissionRequest)? callback =
+              weakReference.target?._onPermissionRequestCallback;
+          if (callback == null) {
+            return request.deny();
+          } else {
+            final Set<WebViewPermissionResourceType> types = request.resources
+                .map<WebViewPermissionResourceType?>((String type) {
+                  switch (type) {
+                    case android_webview.PermissionRequest.videoCapture:
+                      return WebViewPermissionResourceType.camera;
+                    case android_webview.PermissionRequest.audioCapture:
+                      return WebViewPermissionResourceType.microphone;
+                    case android_webview.PermissionRequest.midiSysex:
+                      return AndroidWebViewPermissionResourceType.midiSysex;
+                    case android_webview.PermissionRequest.protectedMediaId:
+                      return AndroidWebViewPermissionResourceType
+                          .protectedMediaId;
+                  }
+
+                  // Type not supported.
+                  return null;
+                })
+                .whereType<WebViewPermissionResourceType>()
+                .toSet();
+
+            if (types.isEmpty) {
```

Comment:
It's probably worth adding a comment explaining that this is to handle the case where the only types requested are unknown.